Feature/favourites

prasenbhandari opened the pull request.

prasenbhandari committed:
Added User Favorite feature

prasenbhandari committed:
fixed favorite button visible to unauthenticated user

prasenbhandari committed:
Merge upstream/development-php-8.3.9 into feature/favourites

ketan404 commented on app/Document.php:
* Rename favouritedByUsers() to favorites(). This way, we should be able to code like this
$user->favorites
* Instead of adding a function isFavotiedBy($user_id), think of adding a function in the Favorite model.
$favorite->isMyFavorite()  // return true/false.
* Remove favoritesCount() ... not required.
One can directly do this in the view/controller like $user->favorites->count().

ketan404 commented on app/User.php:
Like I mentioned above, rename these functions to 
favorites() and remove hasFavorited... I think this should be called on the favorite model like ... $favorite->isMyFavorite()

ketan404 commented on resources/views/layouts/app.blade.php:
Do not add in the layout. Push your favorites.js from only those views where it is required. You can use @push.

ketan404 commented on resources/views/layouts/navbars/navs/user-home.blade.php:
Looks good.

ketan404 commented on routes/web.php:
Typo ?

ketan404 reviewed (commented):
I added comments. Kindly resolve and push your branch.

ketan404 commented on database/migrations/2025_09_30_095759_add_user_favourite_table.php:
Looks good.

ketan404 reviewed (changes requested):
Added comments for multiple files.

prasenbhandari committed:
refactor: Implement PR review

- Rename relationship method
    - User: favoriteDocuments() -> favorites()
    - Document: favoritedByUsers() -> favorites()
    - Remove methods: hasFavorited, isFavoritedBy, favoritesCount

- Add isMyFavorite method to UserFavoritet to check ownership

- Enhance favorites table UX
    - Make document tilts clickable links to documents detail page

- Change favorites.js loading
    - Remove global favorites.js from app.blade.php
    - Add favorites.js to specific page using @push
        - favorites/index.blade.php
        - collection.blade.php

ketan404 commented on app/UserFavorite.php:
isMyFavorite() should not accept anything. You have passed userId.
One more thing... variable names should be like user_id. Use camel case in function names (isMyFavorite).
isMyFavorite(){
return $this->user_id === auth()->id();
}

ketan404 reviewed (changes requested):
A new comment added.

ketan404 reviewed (changes requested):
Minor updates needed. Left comments.

prasenbhandari committed:
minor fixes

ketan404 reviewed (approved):
Good work
